loopchain/tools/config_gen/config_gen.py

```diff
@@ -0,0 +1,196 @@
+
+        channels_config = dict()
+        for channel_name in self._channel_names:
+            channel_config = make_channel_config(
+                crep_root_hash=crep_root_hash,
+                radiostations=radiostations,
+                **kwargs)
+            channels_config[channel_name] = channel_config
+
+        self.channels_config = channels_config
+
+        return channels_config
+
+    @check_param_exist("keys")
+    @check_param_exist("channels_config")
+    def build_peer_configs(self, **kwargs) -> List[Config]:
+        if self.peer_configs:
+            return self.peer_configs
+
+        peer_configs = []
+        for idx, key in enumerate(self.keys):
+            rest_port = make_port_by_order(idx, start_port=7100)
+            peer_config = make_peer_config(channels_config=self.channels_config, port_peer=rest_port, key=key, **kwargs)
+            peer_configs.append(peer_config)
+
+        peer_configs[0] = self._insert_genesis_path_to_first_channel_of_peer(peer_configs[0])
+        self.peer_configs = peer_configs
+
+        return peer_configs
+
+    def _insert_genesis_path_to_first_channel_of_peer(self, peer_config: Config) -> Config:
+        peer_config = copy.deepcopy(peer_config)
+
+        channels_config: Config = peer_config["CHANNEL_OPTION"]
+        for channel_name in channels_config.keys():
+            channels_config[channel_name]["genesis_data_path"] = str(self._config_root / "init_genesis_data.json")
+
+        return peer_config
+
+    @check_param_exist("keys")
+    def build_channel_manage_data(self) -> Config:
+        if not self.channel_manage_data:
+            self.channel_manage_data = make_channel_manage_data(channel_names=self._channel_names, keys=self.keys)
+
+        return self.channel_manage_data
+
+    def write(self):
+        self.write_keys()
+        self.write_genesis_data()
+        self.write_peer_configs()
+        self.write_channel_manage_data()
+        self.write_monitoring_list()
+
+    @check_param_exist("keys")
+    def write_keys(self) -> Iterable[Path]:
+        self._config_root.mkdir(exist_ok=True)
+        self.key_root.mkdir(parents=True)
+        for key in self.keys:
+            key.write()
+
+        return (key.path for key in self.keys)
+
+    @check_param_exist("genesis_data")
+    def write_genesis_data(self) -> Path:
+        self._config_root.mkdir(exist_ok=True)
+
+        genesis_data_path = self._config_root / "init_genesis_data.json"
+        dict_write(path=genesis_data_path, dict_obj=self.genesis_data)
+
+        return genesis_data_path
+
+    @check_param_exist("peer_configs")
+    def write_peer_configs(self) -> Iterable[Path]:
+        self._config_root.mkdir(exist_ok=True)
+
+        paths = []
+        for idx, peer_config in enumerate(self.peer_configs):
+            config_path = self._config_root / f"test_{idx}_conf.json"
+            paths.append(config_path)
+            dict_write(config_path, peer_config)
+
+        return paths
```

Comment:
Why the return type of paths is different between `write_keys` and `write_peer_configs`? If it's intended, I think it needs to be applied in the tests as well.

Reply:
It's my mistake. I changed:
- return `List[Path]`, not Generator[Path]